fluent_pose_synthesis/data/load_data.py

```diff
@@ -0,0 +1,203 @@
+
+        # Generate sliding window indices using vectorized method
+        # Create a list to hold indices arrays for each sample
+        self.clip_indices_list = []
+        sample_idx = 0
+        for i, example in enumerate(self.examples):
+            # Get the number of frames in the fluent sequence; Skip if the sequence is too short
+            min_length = min(example['fluent'].body.data.shape[0], example['disfluent'].body.data.shape[0])
+            if min_length < self.window_size:   
+                continue
+            # Generate sliding window start indices with the given offset
+            clip_indices = np.arange(0, min_length - self.window_size + 1, self.offset_frame)[:, None] + np.arange(self.window_size)
+            # Prepend the sample index (sample_idx) to each clip index row
+            clip_indices_with_idx = np.hstack((np.full((len(clip_indices), 1), i, dtype=clip_indices.dtype), clip_indices))
+            self.clip_indices_list.append(clip_indices_with_idx)
+            sample_idx += 1
+        # Concatenate all clip indices from different samples
+        self.clip_indices = np.concatenate(self.clip_indices_list, axis=0)
+
+        total_clips = len(self.clip_indices)
+        total_frames = sum(frame_nums_list)
+        print('Dataset loaded, trained with %d clips, %d frames, %d mins in total' % (total_clips, total_frames, total_frames/25/60))   # All sequences are sampled at 25fps
+
+    def __len__(self) -> int:
+        """
+        Returns the total number of sliding-window clips in the dataset.
+        """
+        return len(self.clip_indices)
+
+    def __getitem__(self, idx: int) -> Dict[str, Any]:
+        """
+        Retrieves a sliding-window clip from the dataset.
+        Args:
+            idx (int): Index of the clip to retrieve.
+        """
+        # Each row of clip_indices: [example_index, frame0, frame1, ..., frame(window_size-1)]
+        indices = self.clip_indices[idx]
+        ex_idx = indices[0]
+        # The remaining indices are the frame indices for this clip
+        clip_frame_indices = indices[1:]
+        example = self.examples[ex_idx]
+        
+        # Slice the pose sequences using the precomputed indices
+        disfluent_seq = example['disfluent'].body.data[clip_frame_indices].astype(self.dtype)
+        fluent_seq = example['fluent'].body.data[clip_frame_indices].astype(self.dtype)
+        
+        input_clip = disfluent_seq[:self.disfluent_frame]
+        target_clip = fluent_seq[self.disfluent_frame:]
+                
+        # Get the mask for the fluent pose sequence over these frames
+        mask = example['fluent'].body.mask[clip_frame_indices]
+        
+        return {
+            'data': target_clip,
+            'conditions': {
+                'input_clip': input_clip,
+                'mask': mask,
+                'metadata': example['metadata']
+            }
+        }
+
+
+def custom_collate(batch):
```

Comment:
you might want to look into [`zero_pad_collator`](https://github.com/sign-language-processing/pose/blob/7877447d236221fcd0e8705f8017e24eff9250f0/src/python/pose_format/torch/masked/collator.py#L47) which you can import, and it handles stacking, padding, etc...  no need for custom implementation :)

all it expects is for the data to be torch tensors rather than numpy